bindings/c/test/apitester/TesterTransactionExecutor.cpp

```diff
@@ -90,13 +90,15 @@ class TransactionContextBase : public ITransactionContext {
 			fdbDb = executor->selectDatabase();
 		}
 
+		if (tenantName) {
+			fdbTenant = fdbDb.openTenant(*tenantName);
+			fdbDbOps = std::make_shared<fdb::Tenant>(fdbTenant);
+		} else {
+			fdbDbOps = std::make_shared<fdb::Database>(fdbDb);
```

Comment:
I ran a few runs with TSAN it seemed happy enough.

Reply:
I'm a little confused at the use of atomics here, because to me this is thread safe wrt to the continuation executor. Am I missing something?

Reply:
You mean the atomics for replacing the database & transaction objects? These are necessary, because the change can happen in one thread, but certain callbacks of the workload accessing these objects are executed in another thread. Some workloads are executing multiple concurrent reads within the same transaction, the callbacks of which can be scheduled on different threads and executed concurrently, so there are various tricky concurrent scenarios to handle.

Reply:
Yeah it's not super great, now that there are 3 values (db, tenant, dbOps) that all reference the same object, but if they can be intermingled, then you can in theory have a small window where db is updated, but tenant and dbOps are the previous ones.

Reply:
Yes it is not nice. At first it was only db for accessing database operations, now we have also tenant and the dbOps. Maybe once you introduced dbOps, there is no need to access db and tenant directly anymore.